go/tools/gazel/gazel/diff.go

```diff
@@ -0,0 +1,52 @@
+/* Copyright 2016 The Bazel Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"io/ioutil"
+	"os"
+
+	bzl "github.com/bazelbuild/buildifier/core"
+	"github.com/bazelbuild/buildifier/differ"
+)
+
+func diffFile(file *bzl.File) (err error) {
+	f, err := ioutil.TempFile("", "BUILD")
+	if err != nil {
+		return err
+	}
+	defer func() {
+		merr := os.Remove(f.Name())
+		if err == nil {
+			err = merr
+		}
+	}()
+	defer f.Close()
```

Comment:
> In my experience, worrying about the error from the defer is overkill, and creates complexity without much gain.

I agree.

In my experience it was necessary to correctly deal with disk full in the target file system or something. But as you said, worrying about that does not add much value in this specific case of short-living process unlike my experience.

> I would have written (instead of line 31-49):
> ```go
> defer os.Remove(f.Name())
> defer f.Close()
> if _, err := f.Write(bzl.Format(file)); err != nil {
> return err
> }
> ```

done. But I added `Sync` because `Close` used to also imply `Sync` in the original code.